generators/voxel_generator_noise.cpp

```diff
@@ -1,16 +1,21 @@
 #include "voxel_generator_noise.h"
 
+void VoxelGeneratorNoise::set_noise(Ref<OpenSimplexNoise> noise) {
+	_noise = noise;
+}
+
 void VoxelGeneratorNoise::set_channel(VoxelBuffer::ChannelId channel) {
 	ERR_FAIL_INDEX(channel, VoxelBuffer::MAX_CHANNELS);
 	_channel = channel;
+	emit_changed();
```

Comment:
Should emit only if `_channel` actually change

Reply:
done